pkg/apis/stork/v1alpha1/applicationclone.go

```diff
@@ -0,0 +1,86 @@
+package v1alpha1
+
+import (
+	meta "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	//ApplicationCloneResourceName is the name for the application clone resource
+	ApplicationCloneResourceName = "applicationclone"
+	//ApplicationCloneResourcePlural is the name in plural for the application clone resources
+	ApplicationCloneResourcePlural = "applicationclones"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+//ApplicationClone represents the cloning of application in different namespaces
+type ApplicationClone struct {
+	meta.TypeMeta   `json:",inline"`
+	meta.ObjectMeta `json:"metadata,omitempty"`
+	Spec            ApplicationCloneSpec   `json:"spec"`
+	Status          ApplicationCloneStatus `json:"status,omitempty"`
+}
+
+//ApplicationCloneSpec defines the spec to create an application clone
+type ApplicationCloneSpec struct {
+	//NamespaceMapping to store the target:destination namespaces
+	NamespaceMapping map[string]string `json:"namespace_mapping"`
```

Comment:
@sharma-tapas, Dinesh and I were discussing offline about the mapping. We feel we should have a sourceNamespace and destinationNamespace and clone only one namespace at at time.

Reply:
We can probably have the sourceNamespace optional in the code. If specified we will should make sure that it is the same as ApplicationClone namespace unless the AppClone is in admin namespace.

Reply:
Changed the spec to have SourceNamespace and DestinationNamespace, please check the syntax for making the SourceNamespace optional.